rust_src/src/process.rs

```diff
@@ -263,6 +268,15 @@ pub fn process_status(process: LispObject) -> LispObject {
     status
 }
 
+/// Return the connection type of PROCESS.
+/// The value is either the symbol `real', `network', `serial', or `pipe'.
+/// PROCESS may be a process, a buffer, the name of a process or buffer, or
+/// nil, indicating the current buffer's process.
+#[lisp_fn]
+pub fn process_type(process: LispProcessRef) -> LispObject {
+    process.process_type()
```

Comment:
The argument to `process-type` can either be a process/buffer/string/nil. When you define the argument as of type `LispProcessRef`, it accepts only process argument.

Reply:
Had a feeling it wasn't so simple ;) thanks!

Reply:
Indeed. You need the call to `get_process` or its equivalent.